src/Compiler/Facilities/Activity.fs

```diff
@@ -0,0 +1,29 @@
+﻿// Copyright (c) Microsoft Corporation.  All Rights Reserved.  See License.txt in the project root for license information.
+
+namespace FSharp.Compiler.Diagnostics
+
+open System
+open System.Diagnostics
+
+[<RequireQualifiedAccess>]
+module Activity =    
+    
+    let private dummyDisposable =
+        {
+            new IDisposable with 
+                member this.Dispose() = ()
+        }
+    
+    let private activitySourceName = "fsc"
+    let private activitySource = new ActivitySource(activitySourceName)
+
+    let start name (tags : (string * string) seq) : IDisposable = 
+        match activitySource.StartActivity(name) |> Option.ofObj with
+        | Some activity ->
+            for key, value in tags do
+                activity.AddTag(key, value) |> ignore
+            activity :> IDisposable
+        | None ->
+            dummyDisposable
```

Comment:
Please check the PR into your fork.
We can avoid allocating the dummyDisposable, but it is needed for the internal computation expression's .Using to be defensive.
Basically, the .Using was implemented by calling .Dispose() without checking for nullness.

Reply:
>Please check the PR into your fork.

I already did. The problem was that the old code was doing `.AddTag` on a possibly null activity (which I since changed).

You're right that the `dummyDisposable` is not needed (although it's a singleton, so only allocated once).
I modified the code to return `null`. I wasn't aware that `using` (in C#) and `use` (in F#) handle `null`.

Reply:
All good now, I see that also the CE's .Using is in the defensive version now.